tests/test_expressions.py

```diff
@@ -0,0 +1,191 @@
+    """
+    Verify a metrics query, with a label constraint using a regular expression match.
+    """
+    result = prometheus_client.custom_query(query="""
+    prometheus_http_requests_total{
+        job =~ "prom.+"
+    }
+    """)
+    assert len(result) > 15
+
+
+def test_expression_re_success_dots(prometheus_client):
+    """
+    Verify a metrics query, with a label constraint using a regular expression match.
+    """
+    result = prometheus_client.custom_query(query="""
+    prometheus_http_requests_total{
+        code =~ "2.."
+    }
+    """)
+    assert len(result) > 15
+
+
+def test_expression_re_success_or(prometheus_client):
+    """
+    Verify a metrics query, with a label constraint using a regular expression match.
+    """
+    result = prometheus_client.custom_query(query="""
+    prometheus_http_requests_total{
+        code =~ "2..|3.."
+    }
+    """)
+    assert len(result) > 15
+
+
+def test_expression_re_empty(prometheus_client):
+    """
+    Verify a metrics query, with a label constraint using a regular expression match.
+    """
+    result = prometheus_client.custom_query(query="""
+    prometheus_http_requests_total{
+        job =~ "foobar.+"
+    }
+    """)
+    assert len(result) == 0
+
+
+def test_expression_nre_success_basic(prometheus_client):
+    """
+    Verify a metrics query, with a label constraint using a negated regular expression match.
+    """
+    result = prometheus_client.custom_query(query="""
+    prometheus_http_requests_total{
+        job !~ "foobar.+"
+    }
+    """)
+    assert len(result) > 15
+
+
+def test_expression_nre_success_dots(prometheus_client):
+    """
+    Verify a metrics query, with a label constraint using a negated regular expression match.
+    """
+    result = prometheus_client.custom_query(query="""
+    prometheus_http_requests_total{
+        code !~ "5.."
+    }
+    """)
+    assert len(result) > 15
+
+
+def test_expression_nre_empty(prometheus_client):
+    """
+    Verify a metrics query, with a label constraint using a negated regular expression match.
+    """
+    result = prometheus_client.custom_query(query="""
+    prometheus_http_requests_total{
+        job !~ "prom.+"
+    }
+    """)
+    assert len(result) == 0
+
+
+def test_expression_all(prometheus_client):
+    """
+    Verify a metrics query, with label constraints using all types of label matchers.
+    """
+    result = prometheus_client.custom_query(query="""
+    prometheus_http_requests_total{
+        code = "200",
+        handler != "foobar",
+        job =~ "prom.+",
+        instance !~ "foobar",
+    }
+    """)
+    assert len(result) > 15
+
+
+def test_range_vector_selector_success(prometheus_client):
+    """
+    Verify range vector literals.
+
+    https://prometheus.io/docs/prometheus/latest/querying/basics/#range-vector-selectors
+    """
+    result = prometheus_client.custom_query(query="""
+    prometheus_http_requests_total{}[1s]
```

Comment:
Should we use a larger range to avoid risking occasional failures?

Reply:
The risk for flukes on the test suite is far more likely on other spots. I haven't been able to discover any flaws here, yet.

You are referring to the case where this query expression would yield no data, right?

Reply:
I am sorry, I thought this would look at 1 second back from now, but it seems it may look at the last 1 second of data available so it should be fine.

Reply:
Thanks. No need to apologize, we dearly need attentive reviews like yours.